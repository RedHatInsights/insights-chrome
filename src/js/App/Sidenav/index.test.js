import React from 'react';
import ConnectedSideNav, { SideNav }  from './index';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';

describe('SideNav', () => {
    it('should render correctly', () =>{
        const props = {
            activeTechnology: 'someTechnology',
            activeLocation: 'someLocation' };
        const wrapper = shallow(
            <SideNav {...props}/>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

describe('ConnectedSideNav', () =>{
    let initialState;
    let mockStore;

    beforeEach(() =>{
        mockStore = configureStore();
        initialState = ({ chrome: {
            activeTechnology: 'someTechnology',
            activeLocation: 'someLocation'
        } });
    });
    it('should render correctly', () => {
        const store = mockStore(initialState);
        const wrapper = shallow(
            <Provider store={store}>
                <ConnectedSideNav/>
            </Provider>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('should render correctly part 2', () => {
        const store = mockStore(initialState);
        const wrapper = shallow(
            <Provider store={store}>
                <ConnectedSideNav/>
            </Provider>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

});
