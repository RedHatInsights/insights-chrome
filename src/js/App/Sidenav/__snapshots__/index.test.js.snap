// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedSideNav should render correctly 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(SideNav) />
</ContextProvider>
`;

exports[`ConnectedSideNav should render correctly part 2 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(SideNav) />
</ContextProvider>
`;

exports[`SideNav should render correctly 1`] = `
<Fragment>
  <AppSwitcher
    currentApp="someTechnology"
  />
  <Connect(Navigation) />
</Fragment>
`;
